vignettes/Rcpp-modules.Rmd

```diff
@@ -855,6 +855,72 @@ It can be used to perform operations, releasing resources, etc ...
 The finalizer is called automatically when the \proglang{R} object that encapsulates
 the \proglang{C++} object is garbage collected.
 
+### Object factories
+
+The `.factory` member function of `class_` can be used to register a
+[factory](https://en.wikipedia.org/wiki/Factory_method_pattern) that
+can be used as alternative to a constructor.  A factory can be a
+static member function or a free function that returns a pointer to
+the target class. Typical use-cases include creating objects in a
+hierarchy:
+
+```c++
```

Comment:
Very minor nit @rstub, but for consistency could you use: `cpp` instead of `c++`? 

As `cpp` is what is used throughout the rest of the document.

Reply:
Sure thing.